bin/programs/client/src/hint.rs

```diff
@@ -37,7 +37,7 @@ pub enum HintType {
 impl HintType {
     /// Encodes the hint type as a string.
     pub fn encode_with(&self, data: &[&[u8]]) -> String {
-        let concatenated = data.iter().map(hex::encode).collect::<Vec<_>>().join(" ");
+        let concatenated = data.iter().map(hex::encode).collect::<Vec<_>>().join("");
```

Comment:
Since each byte is hex encoded as two hex nibbles, shouldn't we be able to flatten the list and hex encode the entire thing? I wonder if this is more efficient + idiomatic than mapping many `hex::encode` ops and joining strings.

```suggestion
        let concatenated = hex::encode(data.iter().flatten().collect::<Vec<_>>());
```